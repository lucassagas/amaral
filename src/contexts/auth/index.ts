export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
