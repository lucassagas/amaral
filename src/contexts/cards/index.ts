export { CardContext, CardContextProvider } from './cards';

