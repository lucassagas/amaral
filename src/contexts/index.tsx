import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
