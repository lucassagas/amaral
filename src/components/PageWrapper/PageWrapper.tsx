import {
  Box,
  Button,
  Flex,
  Heading,
  Icon,
  Spinner,
  Tooltip,
} from '@chakra-ui/react'
import NextLink from 'next/link'
import { ReactNode } from 'react'
import { RiAddLine, RiRefreshLine } from 'react-icons/ri'
import { spin } from 'src/animations'

type PageWrapperProps = {
  children?: ReactNode
  isLoading?: boolean
  isFetching?: boolean
  refetch: () => void
  title?: string
  registrationRoute?: string
}

export function PageWrapper({
  isFetching,
  isLoading,
  refetch,
  title,
  registrationRoute,
  children,
}: PageWrapperProps) {
  return (
    <Box flex="1" borderRadius={8} bg="gray.800" p="8">
      <Flex mb="8" justify="space-between" align="center">
        <Heading size="lg" fontWeight="normal">
          {title}
          {!isLoading && isFetching && (
            <Spinner size="sm" color="gray.500" ml="4" />
          )}
        </Heading>

        <Flex>
          <Tooltip label="Atualizar">
            <Button
              size="sm"
              fontSize="sm"
              bgColor="gray.700"
              cursor="pointer"
              onClick={refetch}
            >
              <Icon
                as={RiRefreshLine}
                fontSize={20}
                animation={
                  !isLoading && isFetching ? `${spin} 1s infinite linear` : ''
                }
              />
            </Button>
          </Tooltip>
          <NextLink href={registrationRoute!} passHref>
            <Button
              as="a"
              size="sm"
              fontSize="sm"
              colorScheme="orange"
              ml="4"
              leftIcon={<Icon as={RiAddLine} fontSize={20} />}
              cursor="pointer"
            >
              Cadastrar
            </Button>
          </NextLink>
        </Flex>
      </Flex>
      {children}
    </Box>
  )
}
