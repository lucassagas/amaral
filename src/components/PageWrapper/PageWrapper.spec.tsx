import { AppProvider } from '@/contexts/index'
import { render, screen } from '@testing-library/react'
import { PageWrapper } from './PageWrapper'

type PageWrapperProps = {
  isFetching?: boolean
}

const renderComponent = ({ isFetching }: PageWrapperProps) => {
  return render(
    <AppProvider>
      <PageWrapper
        isFetching={isFetching}
        title="Example"
        refetch={() => {}}
        registrationRoute="/example"
      />
    </AppProvider>
  )
}

describe('<PageWrapper />', () => {
  it('should render correctly', () => {
    const { container } = renderComponent({
      isFetching: false,
    })

    expect(container).toMatchSnapshot()
  })

  it('should render spinner when isFetching ', () => {
    renderComponent({ isFetching: true })

    expect(screen.getByText(/loading\.\.\./i)).toBeInTheDocument()
  })
})
