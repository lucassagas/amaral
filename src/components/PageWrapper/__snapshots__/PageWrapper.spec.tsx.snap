// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageWrapper /> should render correctly 1`] = `
<div>
  <div
    class="css-1vfdvw2"
  >
    <div
      class="css-6iav15"
    >
      <h2
        class="chakra-heading css-sxvlx"
      >
        Example
      </h2>
      <div
        class="css-k008qs"
      >
        <button
          class="chakra-button css-fkz59o"
          type="button"
        >
          <svg
            class="chakra-icon css-12tbk7p"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M5.463 4.433A9.961 9.961 0 0 1 12 2c5.523 0 10 4.477 10 10 0 2.136-.67 4.116-1.81 5.74L17 12h3A8 8 0 0 0 6.46 6.228l-.997-1.795zm13.074 15.134A9.961 9.961 0 0 1 12 22C6.477 22 2 17.523 2 12c0-2.136.67-4.116 1.81-5.74L7 12H4a8 8 0 0 0 13.54 5.772l.997 1.795z"
              />
            </g>
          </svg>
        </button>
        <a
          class="chakra-button css-1hvb8bx"
          href="/example"
        >
          <span
            class="chakra-button__icon css-1wh2kri"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-1cv0b1u"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M11 11V5h2v6h6v2h-6v6h-2v-6H5v-2z"
                />
              </g>
            </svg>
          </span>
          Cadastrar
        </a>
      </div>
    </div>
  </div>
</div>
`;
